src/components/Layout/Header/NavBar/NavBar.tsx

```diff
@@ -1,14 +1,16 @@
 import { HStack, StackProps } from '@chakra-ui/react'
+import union from 'lodash/union'
+import { pluginManager } from 'plugins'
 import { useTranslate } from 'react-polyglot'
 import { routes } from 'Routes/Routes'
 
 import { MainNavLink } from './MainNavLink'
-
 export const NavBar = (props: StackProps) => {
   const translate = useTranslate()
+
   return (
     <HStack spacing={12} ml='auto' mr='auto' alignSelf='center' {...props}>
-      {routes
+      {union(routes, pluginManager.getRoutes())
```

Comment:
Instead of pulling this in from a global variable, the `pluginManager` (or perhaps the `pluginRoutes`, which would avoid an extra `useEffect` here but seems less elegant and extensible to me) should be passed down through the whole component stack: `Routes` -> `Layout` -> `Header` -> `HeaderContent` -> `NavBar`.